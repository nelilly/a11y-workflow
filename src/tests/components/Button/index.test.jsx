import React from 'react';
import {shallow, mount} from 'enzyme';
import ReactDOMServer from 'react-dom/server';
// import {axe} from '../../../jest-config/axe-config';
import {axe, toHaveNoViolations} from 'jest-axe';
expect.extend(toHaveNoViolations);

import {Button} from '../../../../src/components/Button/index.jsx';

describe('Button Tests', function() {
  test('Button Snapshop', () => {
    const wrapper = shallow(
      <Button />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Expect no AXE violations', async () => {
    const wrapper = mount(
      <Button />
    );
    expect(await axe(ReactDOMServer.renderToString(wrapper))).toHaveNoViolations();
  });
});
