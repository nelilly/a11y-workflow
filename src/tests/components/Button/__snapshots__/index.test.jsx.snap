// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Button Snapshop 1`] = `
<div>
  <input
    className="button"
    type="button"
    value="Push"
  />
</div>
`;
